src/qibolab/platforms/abstract.py

```diff
@@ -34,7 +35,7 @@ class Qubit:
         Other characterization parameters for the qubit, loaded from the runcard.
     """
 
-    name: str
+    name: Union[str, int]
```

Comment:
It was a request from experimentalists to support strings for qubit names as it helps with the labeling. This is for example used for the 21q chip (#349) where the qubits are grouped to four different readout lines, so we label them as A1, ..., A6, B1, ... B5, C1, ..., C5, D1, ..., D5.

I agree that this adds some maintance overhead and is also assymetric to qibo where qubits have no labels and we just use integers to add gates to a circuit. However, I guess we need to work are way through it.

Reply:
I gave this a thought, particularly because @andrea-pasquale told me that autocalibration does not work with string names, but I could not find an acceptable solution. 

Adding a translation structure is possible, however the main issue is that we need the name to go as it is at the driver level. For example in the 21q chip with QM, we are using string names and is useful to dump the QUA program for debugging. If I translate the strings to integers then the dumped program will have the integers and it will be harder to read if we are working with strings at the external level.

I agree that forcing a single type and avoiding `Union` is always better, however I am not sure what is the best approach here so that it works for everyone. Another possibility would be to drop integers and use only strings (which contain the integers anyway).